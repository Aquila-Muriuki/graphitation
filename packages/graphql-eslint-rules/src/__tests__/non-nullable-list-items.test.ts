import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../non-nullable-list-items";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("non-nullable-list-items", rule, {
  valid: [
    {
      code: `
type SomeType {
  someArray: [Item!]
}`,
    },
  ],
  invalid: [
    {
      code: `
type SomeType {
  someArray: [Item]
}`,
      errors: 1,
    },
  ],
});
