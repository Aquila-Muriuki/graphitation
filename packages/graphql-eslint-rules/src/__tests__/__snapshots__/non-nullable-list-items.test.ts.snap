// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type SomeType {
      2 |   someArray: [Item]
      3 | }

#### ❌ Error

      1 | type SomeType {
    > 2 |   someArray: [Item]
        |              ^^^^^ List items must be not nullable.
      3 | }"
`;
